ota_pal.c

```diff
@@ -75,12 +93,183 @@ static psa_image_id_t xOTAImageID = TFM_FWU_INVALID_IMAGE_ID;
  */
 extern psa_key_handle_t xOTACodeVerifyKeyHandle;
 
+#if !defined( OTA_PAL_CODE_SIGNING_ALGO ) || ( OTA_PAL_CODE_SIGNING_ALGO == OTA_PAL_CODE_SIGNING_ECDSA )
+    static uint8_t ucECDSARAWSignature[ ECDSA_SHA256_RAW_SIGNATURE_LENGTH ] = { 0 };
+#endif /* !defined( OTA_PAL_CODE_SIGNING_ALGO ) || ( OTA_PAL_CODE_SIGNING_ALGO == OTA_PAL_CODE_SIGNING_ECDSA ) */
+
+const AppVersion32_t appFirmwareVersion = { 0 };
+
 /***********************************************************************
  *
  * Functions
  *
  **********************************************************************/
 
+static bool prvConvertToRawECDSASignature( const uint8_t * pucEncodedSignature,  uint8_t * pucRawSignature )
+{
+    bool xReturn = true;
+    const uint8_t * pxNextLength = NULL;
+    uint8_t ucSigComponentLength;
+
+    if( ( pucRawSignature == NULL ) || ( pucEncodedSignature == NULL ) )
+    {
+        xReturn = false;
+    }
+
+    if( xReturn == true )
+    {
+        /* 
+         * The signature has the ASN1-DER format:
```

Comment:
ASN1-DER format is from cloud side. AWS cloud put the signature in ASN1-DER with image.

Reply:
I also tried to put original signature as input to psa_verify_hash, but I got failure return.

Reply:
Sounds weird. The psa_hash_verify() API supports PEM and DER format signatures and I have once tested that. Seems the ota agent does not records algorithm or format information. How about adding a macro like SIGNATURE_ASN1-DER_FORMAT to control whether the conversion is needed so that we can make the code more universal?

Reply:
Configuration OTA_PAL_SIGNATURE_ASN1_DER_FORMAT added to enable convertion.